http/sys_init.go

```diff
@@ -128,3 +135,30 @@ type InitResponse struct {
 type InitStatusResponse struct {
 	Initialized bool `json:"initialized"`
 }
+
+// Validates if the right parameters are used based on AutoUnseal
+func validateInitParameters(core *vault.Core, req InitRequest) error {
+	var recoveryFlagsPresent bool
+	var barrierFlagsPresent bool
+
+	if req.SecretShares != 0 || req.SecretThreshold != 0 || req.StoredShares != 0 || len(req.PGPKeys) != 0 {
+		barrierFlagsPresent = true
+	}
+
+	if req.RecoveryShares != 0 || req.RecoveryThreshold != 0 || len(req.RecoveryPGPKeys) != 0 {
+		recoveryFlagsPresent = true
+	}
+
+	switch core.SealAccess().RecoveryKeySupported() {
+	case true:
+		if barrierFlagsPresent {
```

Comment:
We could be a little more helpful/explicit by specifying the actual param name that we don't like.  Failing that, we could say something in the error like "secret_shares, secret_threshold, and pgp_keys are not applicable to %s" where %s is seal type.